apiserver/worker/compiler.py

```diff
@@ -680,63 +680,139 @@ def get_run_lang(submission_dir):
                     if line[0] == '#':
                         return line[1:-1]
 
-def compile_anything(bot_dir, installTimeLimit=600, timelimit=600, max_error_len = 3072):
+
+INSTALL_ERROR_START = "Possible errors running install.sh. stdout output as follows:"
+INSTALL_ERROR_MID = "End of install.sh stdout. stderr as follows:"
+INSTALL_ERROR_END = "End of install.sh stdout."
+
+
+def truncate_errors(install_stdout, install_errors, language_detection_errors,
+                    compile_errors, max_error_len=10*1024):
+    """
+    Combine lists of errors into a single list under a maximum length.
+    """
+    install_stdout = install_stdout or []
+    install_errors = install_errors or []
+    language_detection_errors = language_detection_errors or []
+    compile_errors = compile_errors or []
+
+    all_errors = language_detection_errors + compile_errors
+    result = []
+    if install_errors:
+        all_errors = install_stdout + install_errors
+
+    if sum(len(line) for line in all_errors) <= max_error_len:
```

Comment:
This isn't including the lengths of langauage_detection and compile errors. I think language detection should always be short but compile errors can be quite long.

Reply:
Oops, misread the code here. It is including language detection and compile errors in all_errors.